src/annotator/notebook.js

```diff
@@ -1,147 +1,131 @@
-    this._prevGroupId = null;
+    this._originalOverflowStyle = this.element.style.overflow;
 
     /**
      * Un-styled shadow host for the notebook content.
-     *
      * This isolates the notebook from the page's styles.
      */
     this._outerContainer = document.createElement('hypothesis-notebook');
     this.element.appendChild(this._outerContainer);
+    const shadowRoot = createShadowRoot(this._outerContainer);
 
-    /**
-     * Lazily-initialized container for the notebook iframe. This is only created
-     * when the notebook is actually used.
-     *
-     * @type {HTMLElement|null}
-     */
-    this.container = null;
-
-    this._emitter.subscribe('openNotebook', groupId => {
-      this._groupId = groupId;
-      this.open();
-    });
-  }
-
-  _update() {
-    const container = this._initContainer();
-
-    // Create a new iFrame if we don't have one at all yet, or if the
-    // groupId has changed since last use
-    const needIframe =
-      !this.frame || !this._prevGroupId || this._prevGroupId !== this._groupId;
-    this._prevGroupId = this._groupId;
-
-    if (needIframe) {
-      this.frame?.remove();
-      this.frame = createNotebookFrame(this.options, this._groupId);
-      container.appendChild(this.frame);
-    }
+    render(
+      <NotebookOuter
+        eventBus={eventBus}
+        config={config}
+        disableScrolling={() => this.disableScrolling()}
+        enableScrolling={() => this.enableScrolling()}
+      />,
+      shadowRoot
+    );
   }
 
-  open() {
-    const container = this._initContainer();
-    this._update();
-    container.classList.add('is-open');
-    container.style.display = '';
-
-    // The overflow CSS property is set to hidden to prevent scrolling of the guest page,
-    // while the notebook is shown as modal. It is restored on the close method.
-    // I believe this hack only works if this.element points to document.body of the guest page.
-    this.originalOverflowStyle = this.element.style.overflow;
+  /**
+   * The overflow CSS property is set to hidden to prevent scrolling of the host page,
+   * while the notebook is shown as modal. It is restored on the close method.
+   * This hack only works if this.element points to document.body of the guest page.
+   */
+  disableScrolling() {
     this.element.style.overflow = 'hidden';
   }
 
-  close() {
-    if (this.container) {
-      this.container.classList.remove('is-open');
-      this.container.style.display = 'none';
-    }
-    this.element.style.overflow = /** @type {string} */ (this
-      .originalOverflowStyle);
+  enableScrolling() {
+    this.element.style.overflow = this._originalOverflowStyle;
   }
 
   destroy() {
     this._outerContainer.remove();
-    this._emitter.destroy();
   }
+}
 
-  _initContainer() {
-    if (this.container) {
-      return this.container;
-    }
+/**
+ * Create the iframe that will load the notebook application.
+ * @param {object} props
+ * @param {Record<string, any>} props.config
```

Comment:
Another nit: these props should be indented. Same goes for other props, uh, props.

Reply:
Thanks.